crucible-mir/src/Mir/Intrinsics.hs

```diff
@@ -861,6 +861,13 @@ data MirStmt :: (CrucibleType -> Type) -> CrucibleType -> Type where
      !(f MirReferenceType) ->
      !(Index ctx tp) ->
      MirStmt f MirReferenceType
+  -- | Like `MirSubfieldRef`, but for fields with statically-unknown types, such
+  -- as trait objects.
+  MirSubfieldRef_Untyped ::
+     !(f MirReferenceType) ->
+     !Int ->
+     !(Maybe (Some TypeRepr)) ->
```

Comment:
Instead of using `Some TypeRepr` here, why not use `TypeRepr tp`, where `tp` is existentially quantified? Better yet, it might help to give `tp` a more descriptive name that indicates what its purpose is in `MirSubfieldRef_Untyped`.

Reply:
Doesn't that require you to choose a `tp` even when the `Maybe` is `Nothing`?

Reply:
I tried this at first, but ran into some errors in the `TestEqualityFC` and `OrdFC` instances for `MirStmt` that I had trouble understanding. I've gone back and tried stubbing out those instances, though, and I don't see errors at calls that provide `Nothing`, so I could probably make that change, if I could work out those instance definitions.

Is one approach considered more idiomatic than the other when working with existentially-quantified types in this way?

Reply:
Ryan and I spoke offline, and figured that the benefit of eliminating `Some` was roughly balanced by the cost of more complicated `{Ord,TestEquality}FC` instance implementations, so I'll leave this as-is.